src/entities/pools/gyro2/gyro2Pool.ts

```diff
@@ -0,0 +1,299 @@
+            const [virtualParamIn, virtualParamOut] = _findVirtualParams(
+                invariant,
+                sqrtAlpha,
+                sqrtBeta,
+            );
+            const inAmountLessFee = _calcInGivenOut(
+                tIn.scale18,
+                tOut.scale18,
+                swapAmount.scale18,
+                virtualParamIn,
+                virtualParamOut,
+            );
+            const inAmount = _addFee(inAmountLessFee, this.swapFee);
+
+            if (mutateBalances) {
+                tIn.decrease(inAmount);
+                tOut.increase(swapAmount.amount);
+            }
+
+            return TokenAmount.fromScale18Amount(tokenIn, inAmount);
+        } catch (_) {
+            return TokenAmount.fromScale18Amount(tokenIn, 0n);
+        }
+    }
+
+    public getLimitAmountSwap(
+        tokenIn: Token,
+        tokenOut: Token,
+        swapKind: SwapKind,
+    ): bigint {
+        const { tIn, tOut, sqrtAlpha, sqrtBeta } = this.getPoolPairData(
+            tokenIn,
+            tokenOut,
+        );
+        if (swapKind === SwapKind.GivenIn) {
+            const invariant = _calculateInvariant(
+                [tIn.scale18, tOut.scale18],
+                sqrtAlpha,
+                sqrtBeta,
+            );
+            const maxAmountInAssetInPool = MathSol.mulUpFixed(
+                invariant,
+                MathSol.divDownFixed(WAD, sqrtAlpha) -
+                    MathSol.divDownFixed(WAD, sqrtBeta),
+            ); // x+ = L * (1/sqrtAlpha - 1/sqrtBeta)
+            const limitAmountIn = maxAmountInAssetInPool - tIn.scale18;
+            const limitAmountInPlusSwapFee = MathSol.divDownFixed(
+                limitAmountIn,
+                WAD - this.swapFee,
+            );
+            return MathSol.mulDownFixed(
+                limitAmountInPlusSwapFee,
+                SWAP_LIMIT_FACTOR,
+            );
+        } else {
+            return MathSol.mulDownFixed(tOut.amount, SWAP_LIMIT_FACTOR);
+        }
+    }
+
+    spotPriceAfterSwapGivenIn(
+        tokenIn: Token,
+        tokenOut: Token,
+        swapAmount: TokenAmount,
+    ): bigint {
+        console.log(tokenIn, tokenOut, swapAmount);
+        throw new Error('Not implemented');
+    }
+
+    spotPriceAfterSwapGivenOut(
+        tokenIn: Token,
+        tokenOut: Token,
+        swapAmount: TokenAmount,
+    ): bigint {
+        console.log(tokenIn, tokenOut, swapAmount);
+        throw new Error('Not implemented');
+    }
+
+    derivativeSpotPriceAfterSwapGivenIn(
+        tokenIn: Token,
+        tokenOut: Token,
+        swapAmount: TokenAmount,
+    ): bigint {
+        console.log(tokenIn, tokenOut, swapAmount);
+        throw new Error('Not implemented');
+    }
+
+    derivativeSpotPriceAfterSwapGivenOut(
+        tokenIn: Token,
+        tokenOut: Token,
+        swapAmount: TokenAmount,
+    ): bigint {
+        console.log(tokenIn, tokenOut, swapAmount);
+        throw new Error('Not implemented');
+    }
```

Comment:
Remove these. This previous concept of swap amount optimization isn't used in `b-sdk`